### PR TITLE
dotCMS/core#19088 Add the template selector and theme field

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/portlets/dot-templates/dot-template-list/components/dot-template-selector/dot-template-selector.component.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                name="template"
+                [(ngModel)]="value"
+                id="designer"
+            />
+            <div class="item">
+                <dot-icon size="100" name="web"></dot-icon>
+                <span>{{ 'templates.template.selector.label.designer' | dm }}</span>
+            </div>
+        </label>
+        <label data-testId="advaced">
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-templates/dot-template-list/components/dot-template-selector/dot-template-selector.component.html:16]: typo "advanced"

**Review by alfredo-dotcms — APPROVED** [review]

File: src/app/portlets/dot-templates/dot-template-create-edit/dot-template-props/dot-template-props.component.ts [hunk 2]
```diff
@@ -23,10 +23,18 @@ export class DotTemplatePropsComponent implements OnInit {
...
+                ...template,
+                title: [template.title, Validators.required],
+                theme: [template.theme, Validators.required]
+            });
+        } else {
+            this.form = this.fb.group({
+                ...template,
+                title: [template.title, Validators.required]
+            });
+        }
```
Comment (ivorpad) [inline src/app/portlets/dot-templates/dot-template-create-edit/dot-template-props/dot-template-props.component.ts:37]: Isn't this more readable? Or is just me?

```javascript
    const formGroupAttrs =
      template.theme !== undefined
        ? {
            ...template,
            title: [template.title, Validators.required],
            theme: [template.theme, Validators.required],
          }
        : {
            ...template,
            title: [template.title, Validators.required],
          };

    this.form = this.fb.group(formGroupAttrs);
```

File: src/app/portlets/dot-templates/dot-template-list/components/dot-template-selector/dot-template-selector.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        });
+
+        it('should have label', () => {
+            const label = advanced.query(By.css('span'));
+            expect(label.nativeElement.innerText).toBe('Advanced');
+        });
+    });
+
+    describe('description', () => {
+        it('should have innterText', () => {
```
Comment (ivorpad) [inline src/app/portlets/dot-templates/dot-template-list/components/dot-template-selector/dot-template-selector.component.spec.ts:131]: innerText*

File: src/app/portlets/dot-templates/guards/dot-template.guard.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                        gotoPortlet: jasmine.createSpy()
+                    }
+                }
+            ]
+        });
+        guard = TestBed.inject(DotTemplateGuard);
+        dotRouterService = TestBed.inject(DotRouterService);
+    });
+
+    it('should be return when path is /advanced', () => {
```
Comment (ivorpad) [inline src/app/portlets/dot-templates/guards/dot-template.guard.spec.ts:27]:
```suggestion
    it('should return true when path is /advanced', () => {
```

File: src/app/portlets/dot-templates/guards/dot-template.guard.spec.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        const segment = new UrlSegment('advanced', null);
+        expect(guard.canLoad(null, [null, segment])).toBe(true);
+    });
+
+    it('should be return when path is /designer', () => {
```
Comment (ivorpad) [inline src/app/portlets/dot-templates/guards/dot-template.guard.spec.ts:32]:
```suggestion
    it('should return true when path is /designer', () => {
```

File: src/app/portlets/dot-templates/guards/dot-template.guard.spec.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        const segment = new UrlSegment('designer', null);
+        expect(guard.canLoad(null, [null, segment])).toBe(true);
+    });
+
+    it('should be return false and redirect with invalid path', () => {
```
Comment (ivorpad) [inline src/app/portlets/dot-templates/guards/dot-template.guard.spec.ts:37]:
```suggestion
    it('should return false and redirect with invalid path', () => {
```

File: src/app/portlets/dot-templates/guards/dot-template.guard.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    canLoad(_route: Route, segments: UrlSegment[]): boolean {
+        const { path } = segments[1];
+
+        if (path === 'designer' || path === 'advanced') {
+            return true;
+        }
+
+        this.dotRouterService.gotoPortlet('templates');
+
+        return false;
```
Comment (ivorpad) [inline src/app/portlets/dot-templates/guards/dot-template.guard.ts:18]: How about?

```javascript
canLoad(_route: Route, segments: UrlSegment[]): boolean {
    const { path } = segments[1];
    this.dotRouterService.gotoPortlet('templates');
    const isValidTemplatePath = path === "designer" || path === "advanced";

    if(!isValidTemplatePath) {
      this.dotRouterService.gotoPortlet('templates');
      return false;
    }

    return isValidTemplatePath;
}
```

Comment (fmontes) [reply src/app/portlets/dot-templates/guards/dot-template.guard.ts:18]: Can't do The `gotoPortlet` is only if is `false`

But I like the `isValidTemplatePath` so I use that

**Pull request merged**